pinot-core/src/main/java/org/apache/pinot/core/query/reduce/AggregationDataTableReducer.java

```diff
@@ -102,6 +103,7 @@ private void reduceWithIntermediateResult(DataSchema dataSchema, Collection<Data
           intermediateResults[i] = _aggregationFunctions[i].merge(mergedIntermediateResult, intermediateResultToMerge);
         }
```

Comment:
Would it make sense to sample and check after merging each aggregation function? Would that become too frequent?

Reply:
Yes. Although generally there shouldn't be that many of aggregation functions, this should change to LoopUtils.sampleAndCheckInterruptionPeriodically for generality.